// Copyright (c) 2021 Huawei Device Co., Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
option optimize_for = LITE_RUNTIME;

message TracePluginConfig {
  repeated string ftrace_events = 1; // kernel event set
  repeated string hitrace_categories = 2; // hitrace event set
  repeated string hitrace_apps = 3; // hitrace app set
  uint32 buffer_size_kb = 4;    // kernel trace buffer size
  uint32 flush_interval_ms = 5; // time interval in milliseconds to notify service process
  uint32 flush_threshold_kb = 6; // buffer water mark threshold to notify service process
  bool parse_ksyms = 7;  // parse /proc/kallsyms or not
  string clock = 8;      // value for trace_clock
  uint32 trace_period_ms = 10; // time interval in milliseconds to read kernel trace buffer
  string raw_data_prefix = 13; // raw data file prefix for debug
  uint32 trace_duration_ms = 20; // time duration in millisconds for trace actions
  bool debug_on = 30; // enable debug options
  bool discard_cache_data = 31; // if set true, session will stop immediately.(cache data will be lost)
  uint32 hitrace_time = 32;
}
